### PR TITLE
Conditional truncating for method name

### PR DESCRIPTION
**Pull request by glafarge**
Prevent base class truncation when not appropriate. See #5770.

### CODE REVIEW

File: src/Error/ExceptionRenderer.php [hunk 1]
```diff
@@ -198,7 +198,11 @@ protected function _customMethod($method, $exception)
     protected function _method(\Exception $exception)
     {
         list(, $baseClass) = namespaceSplit(get_class($exception));
-        $baseClass = substr($baseClass, 0, -9);
+        
+        if(strpos($baseClass, 'Exception')!==false) {
```
Comment (luke83) [inline src/Error/ExceptionRenderer.php]:
```
If (substr($baseClass, -9)==='Exception') {
```

This will avoid bad behavior with `MyExceptionIsStrange`

Comment (glafarge) [inline src/Error/ExceptionRenderer.php]: Fixed ! Thank you @luke83 :+1:

File: src/Error/ExceptionRenderer.php [hunk 2]
```diff
@@ -198,7 +198,11 @@ protected function _customMethod($method, $exception)
     protected function _method(\Exception $exception)
     {
         list(, $baseClass) = namespaceSplit(get_class($exception));
-        $baseClass = substr($baseClass, 0, -9);
+        
+        if(substr($baseClass, -9) === 'Exception') {
```
Comment (dereuromark) [inline src/Error/ExceptionRenderer.php]: Missing space after 'if'.

Comment (glafarge) [inline src/Error/ExceptionRenderer.php]: Fixed. Thank you :+1:

**Comment by jadb** [comment]
Looks good. Can you add a test please? If you need help, let us know.

**Comment by glafarge** [comment]
@jadb Sorry, I know I will have to learn how to deal with tests... For the moment, they are like blurry things, far away from me. Can you help me on it ?

**Comment by markstory** [comment]
I've added a test and merged this in 1d8419943a76df6b112d81f32a670474a10d198f.

**Comment by glafarge** [comment]
Thank you @markstory ! Now I have to understand your tests :)

**Pull request merged**